<a href="https://colab.research.google.com/github/yajuna/tmath308/blob/master/Week6.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# This notebook introduces Chapter 6. You will find a lot of the content familar to you, as we have already seen a lot of the similar topics covered in Chapter 3, in the context of $\mathbb{R}^n$, but now we focus on abstract vector spaces.

As we are already familiar with very similar concepts, we will focus on doing problems using these concepts. Make sure to think and compare what you have seen in Chapter 3, and find what has become more general.

## Definition (vector space)

A set with two operations (addition, scalar multiplication), and satisfy 8 axioms in addition to closure under addition and multiplication.

## Theorem 6.1 (properties regarding zeros and additive inverse)

- $0\mathbf{u}=\mathbf{0}$

- $c\mathbf{0}=\mathbf{0}$

- $(-1)\mathbf{u}=-\mathbf{u}$

- If $c\mathbf{u}=\mathbf{0}$, then $c=0$ or $\mathbf{u}=\mathbf{0}$

## Theorem 6.2 (subspaces)

## Definition (linear combination)

## Definition (span)

## Theorem 6.3 (spans are subspaces)

Exercises: 1, 2, 5, 6, 7, 9, 11, 12, 46, 63, 64

## Section 6.2 studies linear dependence, basis and dimension for abstract vector spaces.

You will see many familiar definitions from Chapter 3. We will again focus on making arguments with definitions and focus on writing.

### Definition (linear independence)
Consider $\{\mathbf{v}_1,...,\mathbf{v}_k\}$, and assume

$c_1\mathbf{v}_1+...+c_k\mathbf{v}_k=\mathbf{0}$.

Then they are linearly dependent if all scalars $c_1,....,c_k$ must be zero.

and linearly independent if _not_ all scalars $c_1,....,c_k$ have to be zero.

### Thm 6.4
$\{\mathbf{v}_1,...,\mathbf{v}_k\}$ is linearly dependent if and only if at least one of them can be expressed as a linear combination of the others.

Sketch of proof: (this is a "if and only if" proof, so we need to prove both directions).



### When studying abstract vector spaces, it is useful to understand the meaning of $c_1\mathbf{v}_1+...+c_k\mathbf{v}_k=\mathbf{0}$. An example is Example 6.25 in your textbook.

Example 6.25. Let $V=P_n$, then the vectors $\{1,x,x^2,...x^n\}$ are linearly independent.

Sketch of proof: Abtract vector $c_1\mathbf{v}_1+...+c_k\mathbf{v}_k=\mathbf{0}$ translated to

$c_0+c_1x+...+c_nx^n=0$ for all $x\in\mathbb{R}$. As we are assuming the equation holds for all real $x$, the equation is true for $n+1$ distinct real numbers, let's call them $\{x_j\}_{j=0}^n$.

Plugging in the values $\{x_j\}_{j=0}^n$ to $c_0+c_1x+...+c_nx^n=0$ gives us $n+1$ linear equations for the unknown coefficients $\{c_j\}_{j=0}^n$ of the form

$c_0+c_1x_j+...+c_nx_j^n=0$.

The coefficient matrix

\begin{equation}
\begin{pmatrix}
1 & x_0 & \cdots & x_0^n\\
1 & x_1 & \cdots & x_1^n\\
\vdots & \vdots & \cdots & \vdots\\
1 & x_n & \cdots & x_n^n\\
\end{pmatrix}
\end{equation}

forms a Vandermonde matrix, and is always invertible with _distinct_ $\{x_j\}_{j=0}^n$. Then the system

$(\text{Vandermonde matrix} | \mathbf{0})$ has only the trivial solution. This means the vectors $\{1,x,x^2,...x^n\}$ are linearly independent.

The above sketch of proof essentially shows one way to prove the Fundamental Theorem of Algebra.

### The Fundamental Theorem of Algebra:
A polynomial of degree $n$ is identically zero, if it has at least $n+1$ zeros.



In [10]:
import numpy as np
# generate a Vandermonde matrix with numpy's vander

# make a list of distinct x values
N = 10
x = np.linspace(1, N, N)
print('List of values x', x)
# make a Vandermonde matrix with the x values
V = np.vander(x)
print('Vandermonde matrix with x', V)

# Show that this Vandermonde matrix is invertible
print('Determinant of V is det(V)=', np.linalg.det(V))

b = np.zeros((N,1))
print('Max of absolute values of solution to Vx = 0 is', np.max(np.abs(np.linalg.solve(V,b))))

List of values x [ 1.  2.  3.  4.  5.  6.  7.  8.  9. 10.]
Vandermonde matrix with x [[1.00000000e+00 1.00000000e+00 1.00000000e+00 1.00000000e+00
  1.00000000e+00 1.00000000e+00 1.00000000e+00 1.00000000e+00
  1.00000000e+00 1.00000000e+00]
 [5.12000000e+02 2.56000000e+02 1.28000000e+02 6.40000000e+01
  3.20000000e+01 1.60000000e+01 8.00000000e+00 4.00000000e+00
  2.00000000e+00 1.00000000e+00]
 [1.96830000e+04 6.56100000e+03 2.18700000e+03 7.29000000e+02
  2.43000000e+02 8.10000000e+01 2.70000000e+01 9.00000000e+00
  3.00000000e+00 1.00000000e+00]
 [2.62144000e+05 6.55360000e+04 1.63840000e+04 4.09600000e+03
  1.02400000e+03 2.56000000e+02 6.40000000e+01 1.60000000e+01
  4.00000000e+00 1.00000000e+00]
 [1.95312500e+06 3.90625000e+05 7.81250000e+04 1.56250000e+04
  3.12500000e+03 6.25000000e+02 1.25000000e+02 2.50000000e+01
  5.00000000e+00 1.00000000e+00]
 [1.00776960e+07 1.67961600e+06 2.79936000e+05 4.66560000e+04
  7.77600000e+03 1.29600000e+03 2.16000000e+02 3.60000000e+01
  6.00

Exercises
6 (look at Example 6.28),
10 (look at Example 6.24),
11

### Definition (Basis)
A subset $B$ of $V$ is a basis of $V$ if it is both linearly independent and spans $V$.

Exercises 18, 22

### Deifinition (Coordinates with respect to a basis)

If $\mathbf{u}=c_1\mathbf{v}_1+...+c_k\mathbf{v}_k=\mathbf{0}$, then the vector $[\mathbf{u}]|_{B} = (c_1,...,c_k)$ is the coordinate vector of $\mathbf{u}$ with respect to $B=\{\mathbf{v}_1,...,\mathbf{v}_k\}$.

Exercise 26, 28

Note: linearity implies Theorem 6.6, which says $[c\mathbf{u}+\mathbf{v}]|_{B}=c[\mathbf{u}]|_{B}+[\mathbf{v}]|_{B}$

### Theorem 6.7 (Representation of vectors with $\mathbb{R}^n$).

Let $B =\{\mathbf{v}_1,...,\mathbf{v}_n\}$ be a basis for a vector space $V$ and let $\mathbf{u}_1,...,\mathbf{u}_k$
be vectors in $V$. Then $\mathbf{u}_1,...,\mathbf{u}_k$ is linearly independent in $V$ if and only if
$\{[\mathbf{u}_1]|_B,...,[\mathbf{u}_k]|_B\}$ is linearly independent in $\mathbb{R}^n$.

### Exercise 32 shows that span property of $B$ also translates to the coordinate vectors.

Let $B =\{\mathbf{v}_1,...,\mathbf{v}_n\}$ be a basis for a vector space $V$ and let $\mathbf{u}_1,...,\mathbf{u}_k$
be vectors in $V$. Then $\mathbf{u}_1,...,\mathbf{u}_k$ span $V$ if and only if
$\{[\mathbf{u}_1]|_B,...,[\mathbf{u}_k]|_B\}$ span $\mathbb{R}^n$.

Note: These nice properties that translate between $n$ dimensional abstract vector spaces and $\mathbb{R}^n$ is a big reason why we study linear algebra. This is one example of [Representation Theory](https://en.wikipedia.org/wiki/Representation_theory). A nice book to read about these topics (and this point of view) is Sheldon Axler's _Linear Algebra Done Right_.

### Theorem 6.10' (span, linear dependence, and dimension)
Let $V$ be a vector space. $B$ be a set of $n$ vectors. Two of the following implies the last.

1. $B$ is linearly independent.
2. $B$ span the whole space $V$.
3. dim$(V)=n$.